### ISSUE TITLE
the led roulette example misbehaves when compiled in release mode (+LTO)

### ISSUE THREAD

**Issue by japaric** [japaric/f3#23]
For example, this variation of the original example:

``` rust
#[export_name = "main"]
pub extern "C" fn main() -> ! {
    loop {
        for led in LEDS.iter() {
            led.on();
            delay::ms(100);
            led.off();
            delay::ms(100);
        }
    }
}
```

when compiled in debug mode, produces the expected result: _all_ the LEDs light on in a circular cycle.

But if the code is compiled in release mode, then only three, the "first" three, LEDs participate in the "circular" cycle.

This is the only example that misbehaves when compiled in release mode. It also happens to be only example that uses interrupts so perhaps that's the cause of the problem.

**Comment by japaric** [comment]
The problem seems to be the `delay::ms` function. Let's try rolling that function back to use busy polling instead of interrupts.

**Comment by japaric** [comment]
fixed in #31 by roll backing the implementation
